pkg/sensors/tracing/generickprobe.go

```diff
@@ -310,8 +310,11 @@ func preValidateKprobes(name string, kprobes []v1alpha1.KProbeSpec) error {
 			if !bpf.HasOverrideHelper() {
 				return fmt.Errorf("Error override action not supported, bpf_override_return helper not available")
 			}
-			if !f.Syscall && strings.HasPrefix(f.Call, "security_") == false {
-				return fmt.Errorf("Error override action can be used only with syscalls and security_ hooks")
+			// If not a system call and not a security_ hook on minimum kernel 5.7 fail the override
+			// https://lore.kernel.org/all/20200304191853.1529-5-kpsingh@chromium.org/
+			if !f.Syscall &&
+				(strings.HasPrefix(f.Call, "security_") == false || !kernels.MinKernelVersion("5.7")) {
+				return fmt.Errorf("Error override action can be used only with syscalls or on security_ hooks with a minimum kernel version 5.7")
```

Comment:
Until we have a feature check, would it make sense to have a flag so that we can ignore the kernel check here? As @olsajiri said, some vendor kernels backport many patches and the kernel version does not necessarily reflect the feature set of the kernel, and it would be nice to have a way to override the kernel check.

We can leave this as a follow up though.